ydb/library/yql/providers/s3/provider/yql_s3_provider.h

```diff
@@ -27,6 +29,8 @@ struct TS3State : public TThrRefBase
     TS3Configuration::TPtr Configuration = MakeIntrusive<TS3Configuration>();
     const NKikimr::NMiniKQL::IFunctionRegistry* FunctionRegistry = nullptr;
     ISecuredServiceAccountCredentialsFactory::TPtr CredentialsFactory;
+    IHTTPGateway::TPtr Gateway;
+    TKikimrConfiguration::TPtr KikimrConfig;
```

Comment:
Why do we need the kikimr configuration here? We usually have another deps from kqp to library/yql. Doest it correct here?

Reply:
It is used to get MaxTasksPerStage, I can replace this with passing just this option